pkg/asset/installconfig/openstack/openstack.go

```diff
@@ -13,19 +20,161 @@ const (
 	defaultVPCCIDR = "10.0.0.0/16"
 )
 
+// Read the valid cloud names from the clouds.yaml
+func getCloudNames() ([]string, error) {
+	clouds, err := clientconfig.LoadCloudsYAML()
+	if err != nil {
+		return nil, err
+	}
+	i := 0
+	cloudNames := make([]string, len(clouds))
+	for k := range clouds {
+		cloudNames[i] = k
```

Comment:
Why not just use append instead of keeping with an index?

Reply:
Yeah we could do it either way, I read that this approach is slightly more efficient because you don't have to do reallocations, but honesty I'm fine with either.  Honestly I'm amazed there's no native .keys() equivalent and I just found this approach while looking for a go alternative.